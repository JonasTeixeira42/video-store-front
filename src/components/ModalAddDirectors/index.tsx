import React, { FormEvent, useCallback, useState } from 'react';
import { AddBox } from '@styled-icons/material-outlined/';

import Modal, { ModalProps } from 'components/Modal';
import TextField from 'components/TextField';
import Heading from 'components/Heading';
import Form from 'components/Form';

import * as S from 'components/ModalStyles';
import api from 'services/api';

const initialData = {
  name: '',
};

export type ModalAddDirectorsProps = Omit<ModalProps, 'children'>;

const ModalAddDirectors = ({
  isOpen = false,
  setIsOpen,
}: ModalAddDirectorsProps) => {
  const [formData, setFormData] = useState({ name: '' });
  const [resetValue, setResetValue] = useState(false);

  const handleInput = useCallback(
    (inputName: string, inputValue: string) => {
      setFormData({
        ...formData,

        [inputName]: inputValue,
      });
    },
    [formData],
  );

  const handleSubmit = useCallback(
    async (event: FormEvent<HTMLFormElement>) => {
      event.preventDefault();

      try {
        await api.post('directors', formData);

        setResetValue(true);

        alert('Sucess');
      } catch (error) {
        alert(error.response.data.error);
      }
    },
    [formData],
  );

  const handleResetValue = () => {
    setResetValue(!resetValue);
  };

  return (
    <S.Wrapper aria-hidden={!isOpen}>
      <Modal isOpen={isOpen} setIsOpen={setIsOpen}>
        <Heading lineLeft lineColor="primary">
          Add a Director
        </Heading>
        <Form
          buttonLabel="Add Director"
          buttonIcon={<AddBox size={24} />}
          onSubmit={handleSubmit}
        >
          <TextField
            label="Name"
            placeholder="Name"
            name="name"
            resetValue={resetValue}
            setResetValue={handleResetValue}
            initialValue={formData.name}
            onInput={handleInput}
          />
        </Form>
      </Modal>
    </S.Wrapper>
  );
};

export default ModalAddDirectors;
