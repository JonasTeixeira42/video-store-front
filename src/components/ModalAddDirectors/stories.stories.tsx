import { Story, Meta } from '@storybook/react/types-6-0';

import ModalAddDirectors, { ModalAddDirectorsProps } from '.';

export default {
  title: 'ModalAddDirectors',
  component: ModalAddDirectors,
  args: {
    isOpen: true,
  },
} as Meta;

export const Default: Story<ModalAddDirectorsProps> = (args) => (
  <ModalAddDirectors {...args} />
);
