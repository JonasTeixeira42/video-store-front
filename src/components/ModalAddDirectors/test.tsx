import { screen } from '@testing-library/react';
import { renderWithTheme } from 'utils/tests/helpers';

import ModalAddDirectors from '.';

describe('<ModalAddDirectors />', () => {
  it('should render the Modal', () => {
    const { container } = renderWithTheme(<ModalAddDirectors isOpen={true} />);

    const firsChild = container.firstElementChild;

    expect(firsChild?.getAttribute('aria-hidden')).toBe('false');
    expect(
      screen.getByRole('heading', { name: /Add a Director/i }),
    ).toBeInTheDocument();
    expect(screen.getByPlaceholderText(/name/i)).toBeInTheDocument();
    expect(
      screen.getByRole('button', { name: /Add Director/i }),
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });
});
