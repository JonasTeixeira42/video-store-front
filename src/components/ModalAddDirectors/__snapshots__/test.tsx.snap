// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ModalAddDirectors /> should render the Modal 1`] = `
.c0 .c1 {
  margin-bottom: 3.2rem;
}

@media (min-width:768px) {

}

<div
  aria-hidden="false"
  class="c0"
/>
`;
